clusterloader2/framework/framework.go

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package framework
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/dynamic"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/perf-tests/clusterloader2/api"
+	"k8s.io/perf-tests/clusterloader2/framework/util"
+)
+
+const (
+	automanagedNamespaceName = "namespace"
+	contentType              = "application/vnd.kubernetes.protobuf"
+	qps                      = 100
+	burst                    = 200
+)
+
+// Framework allows for interacting with Kubernetes cluster via
+// official Kubernetes client.
+type Framework struct {
+	automanagedNamespaceCount int
+	// TODO(krzysied): Add support for having multiple instances of clients.
+	clientSet     clientset.Interface
+	dynamicClient dynamic.Interface
+}
+
+// NewFramework creates new framework based on given kubeconfig.
+func NewFramework(path string) (*Framework, error) {
+	config, err := util.LoadConfig(path)
+	if err != nil {
+		return nil, fmt.Errorf("config load failed: %v", err)
+	}
+	config.ContentType = contentType
+	config.QPS = qps
+	config.Burst = burst
+
+	clientSet, err := clientset.NewForConfig(config)
+	if err != nil {
+		return nil, fmt.Errorf("creating clientset failed: %v", err)
+	}
+	dynamicClient, err := dynamic.NewForConfig(config)
+	if err != nil {
+		return nil, fmt.Errorf("creating dynamic config failed: %v", err)
+	}
+	return &Framework{0, clientSet, dynamicClient}, nil
+}
+
+// CreateAutomanagedNamespaces creates automanged namespaces.
+func (f *Framework) CreateAutomanagedNamespaces(namespaceCount int) ([]string, error) {
+	if f.automanagedNamespaceCount != 0 {
+		return nil, fmt.Errorf("Automanaged namespaces already created")
+	}
+	var namespaces []string
+	for i := 0; i < namespaceCount; i++ {
+		name := fmt.Sprintf("%v-%d", automanagedNamespaceName, i)
+		if err := util.CreateNamespace(f.clientSet, name); err != nil {
+			return nil, err
+		}
+		namespaces = append(namespaces, name)
+		f.automanagedNamespaceCount++
+	}
+	return namespaces, nil
```

Comment:
For every object related operation there is namespace name required. I though that passing list of names is better solution than making automaneged namepsace names by convention (in other packages than framework).

Other solution could be making public function for generating automanaged namespaces name.

Reply:
Those will be coming with NamespaceRange - so it will be needed in exactly one other place - defaulting function for NamespaceRange. I don't think that returning names here gives us somethimg.

Reply:
Ok. I'm removing namespaces names returning.

I guess at some point in the future we will need to make automanagedNamespaceName const public then, but for now it is not required.